src/core/unit.c

```diff
@@ -5340,6 +5340,7 @@ int unit_set_exec_params(Unit *u, ExecParameters *p) {
                 return r;
 
         p->runtime_scope = u->manager->runtime_scope;
+        p->virtualization = detect_virtualization();
```

Comment:
urks, i am not sure how much i like this, given that there's not just "the one virtualization" in effect, i.e. we have detect_vm() and detect_container() and so on, but his field here only caches detect_virtualization()...

this frankly seems like premature optimization to me. Can we please drop this passing over?

Reply:
Uh? detect_virtualization() calls detect_container first already, and the Virtualization enum can already list both containers and vms. And it does the right thing, because we want to detect lxc, whether in a VM or on bare metal.

Reply:
Also I put this next to "runtime_scope" intentionally. That's also fixed information about the runtime environment that is passed over, I don't really see any difference?

Reply:
if you run in a container inside a VM, then detect_vm() and detect_container() will result in different responses, and detect_virtualization() only returns one of them.

Hence, depending on the usecase we tend to call one of the three, but they are not redundant, and there's a reason why we have three separate calls.

Reply:
Or rather, to keep it smaller, given what I care about is the container manager, I'll rename the variable and check explicitly with that. If the VMM is also needed in the future, it can be added.

Reply:
Of course the alternative is to only check for apparmor, but it seemed and much better to check for the most precise condition possible, given it's easy and cheap